js/app/build_plugins.ts

```diff
@@ -109,18 +87,7 @@ export function generate_dev_entry({ enable }: { enable: boolean }): Plugin {
 }
 
 function make_entry(script: string): string {
-	const make_script = `
-function make_script(src) {
-    const script = document.createElement('script');
-    script.type = 'module';
-    script.setAttribute("crossorigin", "");
-    script.src = src;
-    document.head.appendChild(script);
-}`;
-
-	return `
-${make_script}
-make_script("${script}");
+	return `import("${script}");
```

Comment:
I dunno why i chose the hard route previsouly.

The problem we always had was that anything that is requested from an html tag will always treat its own location as the base url from which to resolve files. With `import` it will use the files own location.

This `gradio.js` file will be request like: `https://aws-etc.com/4.0.0/gradio.js` so when we request`./assets/index-asdasdasd.js` it will be considered relative to that `gradio.js` url. 

When we add a script to the page the 'file' in that case is the HTML file, which isn't where the cdn assets live.